File: 819. Most Common Word/solution.cpp
class Solution {
public:
  string mostCommonWord(string paragraph, vector<string>& banned) {
    unordered_set<string> bannedSet = unordered_set<string>(banned.begin(),banned.end());
    unordered_map<string, int> dictionary = {};
    string word = "";

    for (int index = 0; index < paragraph.size(); index++){
      char letter = paragraph[index];
      if (letter != '!' && letter != '?' && letter != '\'' && letter != ',' && letter != ';' && letter != '.' && letter != ' '){
        word += tolower(letter);
      }
      if (letter == '!' || letter == '?' || letter == '\'' || letter == ',' || letter == ';' || letter == '.' || letter == ' '  || index == paragraph.size() - 1){
        if (word == ""){
          continue;
        }
        if (bannedSet.count(word) != 1){
          dictionary[word]++;
        }
        word = "";
      }
    }

    string maxKey = "";
    int maxValue = 0;
    
    for (auto word : dictionary){
      if (word.second > maxValue){
        maxKey = word.first;
        maxValue = word.second;
      }
    }

    return maxKey;
  }
};

File: 819. Most Common Word/solution.py
class Solution(object):
  def mostCommonWord(self, paragraph, banned):
    """
    :type paragraph: str
    :type banned: List[str]
    :rtype: str
    """
    bannedSet = set(banned)
    dictionary = {}
    word = ""
    
    for index, letter in enumerate(paragraph):
      if letter != "!" and letter != "?" and letter != "'" and letter != "," and letter != ";" and letter != "." and letter != " ":
        word += letter
      if letter == "!" or letter == "?" or letter == "'" or letter == "," or letter == ";" or letter == "." or letter == " "  or index == len(paragraph) - 1:
        if word == "":
          continue
        word = word.lower()
        if word not in bannedSet: 
          if not word in dictionary:
            dictionary[word] = 1
          else:
            dictionary[word] += 1
        word = ""
      
    maxKey = ""
    maxValue = 0
    
    for key in dictionary:
      if dictionary[key] > maxValue:
        maxKey = key
        maxValue = dictionary[key]
        
    return maxKey